### PR TITLE
Better Exception Handling (HttpClient)

### PR DESCRIPTION
**Pull request by jorg3lopez**
# Better Exception Handling for HttpClient

This PR introduces a new custom exception for the `HttpClient` interface. The new exception is triggered in the `Apacheclient` class whenever it gets an HTTP exception from a POST request.

## Issue
#84

### CODE REVIEW

File: app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSenderTest.groovy [hunk 1]
```diff
@@ -45,7 +46,7 @@ class ReportStreamLabOrderSenderTest extends Specification {
         ReportStreamLabOrderSender.getInstance().sendRequestBody("message_2", "fake token")
 
         then:
-        noExceptionThrown()  //This test to be updated whenever the actual code's TODO is addressed for the exception handling
+        thrown(Exception)  //This test to be updated whenever the actual code's TODO is addressed for the exception handling
```
Comment (jorg3lopez) [inline app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSenderTest.groovy:49]: This unit test needs to be revisit once all the exception handling is done.

**Review by halprin — APPROVED** [review]
Rockin'!

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/wrappers/HttpClientException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package gov.hhs.cdc.trustedintermediary.wrappers;
+
+import java.io.IOException;
+
+public class HttpClientException extends IOException {
```
Comment (halprin) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/wrappers/HttpClientException.java:5]: Please add JavaDocs.

**Review by halprin — COMMENTED** [review]
Oh, please make a dedicated unit test for the new exception.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/wrappers/HttpClientException.java (hunk 2 above)
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/wrappers/HttpClientException.java:5]: Totally, agree.

**Review by halprin — APPROVED** [review]
Awe yeah!

**Pull request merged**